src/common/options/rgw.yaml.in

```diff
@@ -3785,3 +3785,15 @@ options:
   flags:
   - startup
   with_legacy: true
+- name: rgw_lua_max_memory_per_state
+  type: int
+  level: advanced
+  desc: Max size of memory used by a single lua state
+  long_desc: This is the maximum size in bytes that a lua state can allocate for its own use.
+    Note that this does not include any memory that can be accessed from lua, but managed by the RGW.
+    If not set, it would use a default of 500K. If set to zero or negative number, the amount of
+    memory would only be limited by the system.
+  default: 500000
```

Comment:
500K seems like a lot?

Reply:
there is no pre-alloation. this is just a maximum.
if we allow for max 1K concurrent requests, there could be a maximum of 500MB used by lua.
i will try with one of the more realistic payloads (that inport packages) and see how much is actually used. will set that accordingly

Reply:
@mattbenjamin i tested the elasticsearch example: https://github.com/ceph/ceph/blob/main/examples/rgw/lua/elasticsearch_adapter.md
and it uses ~300KB.
So, I guess that 500KB is a reasonable default. although, with 10K concurrent session with 500KB is 5GM RAM - which is inot small
since most o the memory is taken by the imported packages (cjson, elasticsearch etc.) we probably need to add an option where we load the packages to a common lua VM, and let all per-request threads access them (assuming they are not storing global state).
will add that as an enhacement to the "lua hardening" effort.
* this would allow us to use a much smaller minimum (probably 100KB)
* less local disk IO for reading the libraries